src/Plugins/ComplexCore/src/ComplexCore/Filters/RawBinaryReaderFilter.cpp

```diff
@@ -0,0 +1,203 @@
+#include "RawBinaryReaderFilter.hpp"
+
+#include "complex/DataStructure/DataPath.hpp"
+#include "complex/Filter/Actions/CreateArrayAction.hpp"
+#include "complex/Filter/Actions/EmptyAction.hpp"
+#include "complex/Parameters/ArrayCreationParameter.hpp"
+#include "complex/Parameters/ChoicesParameter.hpp"
+#include "complex/Parameters/FileSystemPathParameter.hpp"
+#include "complex/Parameters/NumberParameter.hpp"
+#include "complex/Parameters/NumericTypeParameter.hpp"
+
+#include "ComplexCore/Filters/Algorithms/RawBinaryReader.hpp"
+
+#include <filesystem>
+namespace fs = std::filesystem;
+
+using namespace complex;
+
+namespace complex
+{
+
+constexpr int32_t k_RbrZeroComponentsError = -391;
+constexpr int32_t k_RbrNumComponentsError = -392;
+constexpr int32_t k_RbrWrongType = -393;
+constexpr int32_t k_RbrEmptyFile = -394;
+constexpr int32_t k_RbrSkippedTooMuch = -395;
+
+//------------------------------------------------------------------------------
+std::string RawBinaryReaderFilter::name() const
+{
+  return FilterTraits<RawBinaryReaderFilter>::name.str();
+}
+
+//------------------------------------------------------------------------------
+std::string RawBinaryReaderFilter::className() const
+{
+  return FilterTraits<RawBinaryReaderFilter>::className;
+}
+
+//------------------------------------------------------------------------------
+Uuid RawBinaryReaderFilter::uuid() const
+{
+  return FilterTraits<RawBinaryReaderFilter>::uuid;
+}
+
+//------------------------------------------------------------------------------
+std::string RawBinaryReaderFilter::humanName() const
+{
+  return "Raw Binary Importer";
+}
+
+//------------------------------------------------------------------------------
+std::vector<std::string> RawBinaryReaderFilter::defaultTags() const
+{
+  return {"#IO", "#Input", "#Read", "#Import"};
+}
+
+//------------------------------------------------------------------------------
+Parameters RawBinaryReaderFilter::parameters() const
+{
+  Parameters params;
+  // Create the parameter descriptors that are needed for this filter
+  params.insert(std::make_unique<FileSystemPathParameter>(k_InputFile_Key, "Input File", "", fs::path(), FileSystemPathParameter::PathType::InputFile));
+  params.insert(std::make_unique<NumericTypeParameter>(k_ScalarType_Key, "Scalar Type", "", NumericType::int8));
+  params.insert(std::make_unique<Int32Parameter>(k_NumberOfComponents_Key, "Number of Components", "", 0));
```

Comment:
```suggestion
  params.insert(std::make_unique<UInt64Parameter>(k_NumberOfComponents_Key, "Number of Components", "", 0));
```
This should be `uint64` now in complex to match the changes we made in the new DataArray.